Boa.Constrictor/Screenplay/Conditions/Collection/HasSize.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Boa.Constrictor.Screenplay
+{
+    /// <summary>
+    /// Condition to check if the collection size satisfies another condition.
+    ///
+    /// Sample usage:
+    /// Actor.WaitsUntil(TextList.For(ResultPage.ResultLinks), IsAcollectionOfType{string}.AndHasSizeThat(IsEqualTo.Value(10)));
+    /// </summary>
+    public class HasSize<T> : ICondition<IEnumerable<T>>
```

Comment:
To match Linq names, it might be prudent to name this Condition after "Count" instead of "Size".